pkg/synthetictests/duplicated_events.go

```diff
@@ -199,13 +237,94 @@ func (d duplicateEventsEvaluator) testDuplicatedEvents(testName string, flakeOnl
 
 		if flake || flakeOnly {
 			flakes = append(flakes, appendToFirstLine(msg, " result=allow "))
+			fmt.Println("Found patho flake")
 		} else {
+			fmt.Println("Found patho failure")
 			failures = append(failures, appendToFirstLine(msg, " result=reject "))
 		}
+
+		pathoEvent := pathoItem.pathologicalEvent
+		fmt.Println("Checking pathoEvent.Message: ", pathoEvent.Message)
+
+		if strings.Contains(pathoEvent.Message, duplicateevents.PathologicalMark) && !strings.Contains(pathoEvent.Message, duplicateevents.InterestingMark) {
+
+			fmt.Println("Processing event with pathologicalTrue but no interesting mark")
+
+			// This event occurred more than threshold but is not recognized (i.e., unknown).
+			// So we need to mark all other instances of this event so we can chart when this
+			// event started happening.
+
+			clientSet, clientErr := kubernetes.NewForConfig(kubeClientConfig)
+			if clientErr != nil {
+				fmt.Printf("Unexpected error getting AdminConfig for event-raw-events: %v\n", clientErr)
+				fmt.Println("Checking for previous pathological events will be skipped")
+			} else {
+				// Narrow the search by namespace.  For some events (e.g., node related), there
+				// will be no namespace and so we end up searching all events.
+				pathoEventNamespace := monitorapi.NamespaceFromLocator(pathoEvent.Locator)
+				coreEvents, coreEventsErr := clientSet.CoreV1().Events(pathoEventNamespace).List(context.TODO(), metav1.ListOptions{})
+				if coreEventsErr != nil {
+					fmt.Printf("Unexpected error getting events from %s: %v\n", pathoEventNamespace, coreEventsErr)
+					fmt.Println("Going back through events will be skipped")
+				} else {
+					fmt.Printf("Going back through events for %s with length %d\n", pathoEventNamespace, len(coreEvents.Items))
+					matchCount := 0
+					for i, coreEvent := range coreEvents.Items {
+						coreEventObjName := "initially empty match"
+						if len(coreEvent.InvolvedObject.Name) > 0 {
+							coreEventObjName = coreEvent.InvolvedObject.Name
+						}
+
+						originalPathoMessage := originalMessageFromMessage(pathoEvent.Message)
+
+						// Use the same algorithm used in RecordAddOrUpdateEvent to get the From/To time.
+						coreEventFrom := coreEvent.LastTimestamp.Time
+						if coreEventFrom.IsZero() {
+							coreEventFrom = coreEvent.EventTime.Time
+						}
+						if coreEventFrom.IsZero() {
+							coreEventFrom = coreEvent.CreationTimestamp.Time
+						}
+						coreEventTo := coreEventFrom.Add(1 * time.Second)
+
+						fmt.Println("Counter: ", i)
```

Comment:
I'm starting to lean towards use of logrus, I don't dare try in the sub-test processes because we explicitly try to prevent logging from current tests overlapping, but for logging we know is in the main openshift-tests process it should be safe and would be helpful to see contextual logs and consistent timestamps. I'd suggest logrus.Debug and logrus.Info used here as appropriate given the amount of logging you have going on here.

Reply:
Yes, logrus looks nice.